addons/website_sale_delivery/controllers/main.py

```diff
@@ -193,17 +193,28 @@ def express_checkout_process_shipping_address(self, partial_shipping_address):
                     name=_('Anonymous express checkout partner for order %s', order_sudo.name),
                 )
 
-        # Returns the list of develivery carrier available for the sale order.
+        # Returns the list of delivery carrier available for the sale order.
         return sorted([{
-            'id': carrier.id,
-            'name': carrier.name,
-            'description': carrier.website_description,
-            'minorAmount': payment_utils.to_minor_currency_units(
-                WebsiteSaleDelivery._get_rate(carrier, order_sudo, is_express_checkout_flow=True)['price'],
-                order_sudo.currency_id,
-            ),
-        } for carrier in order_sudo._get_delivery_methods()],
-        key=lambda carrier: carrier['minorAmount'])
+                'id': carrier.id,
+                'name': carrier.name,
+                'description': carrier.website_description,
+                'minorAmount': payment_utils.to_minor_currency_units(price, order_sudo.currency_id),
+            } for carrier, price in WebsiteSaleDelivery._get_carriers_express_checkout(order_sudo).items()
+            ], key=lambda carrier: carrier['minorAmount'])
+
+    @staticmethod
+    def _get_carriers_express_checkout(order):
```

Comment:
This method receives (and needs) an order sudoed because public users don't have the rights to compute rate shipment. Let's be clear about that :)
```suggestion
    def _get_carriers_express_checkout(order_sudo):
```